test/autorest/generated/paginggroup/pollers.go

```diff
@@ -28,103 +27,41 @@ type productResultPagerPoller struct {
 }
 
 func (p *productResultPagerPoller) FinalResponse(ctx context.Context) (ProductResultPager, error) {
-	if !p.Done() {
-		return nil, errors.New("cannot return a final response from a poller in a non-terminal state")
-	}
-	if p.pt.pollerMethodVerb() == http.MethodPut || p.pt.pollerMethodVerb() == http.MethodPatch {
-		return p.handleResponse(p.pt.latestResponse())
-	}
-	// checking if there was a FinalStateVia configuration to re-route the final GET
-	// request to the value specified in the FinalStateVia property on the poller
-	err := p.pt.setFinalState()
+	s := &productResultPager{}
+	resp, err := p.pt.FinalResponse(ctx, p.pipeline, s)
 	if err != nil {
 		return nil, err
 	}
-	if p.pt.finalGetURL() == "" {
-		// we can end up in this situation if the async operation returns a 200
-		// with no polling URLs.  in that case return the response which should
-		// contain the JSON payload (only do this for successful terminal cases).
-		if lr := p.pt.latestResponse(); lr != nil && p.pt.hasSucceeded() {
-			result, err := p.handleResponse(lr)
-			if err != nil {
-				return nil, err
-			}
-			return result, nil
-		}
-		return nil, errors.New("missing URL for retrieving result")
-	}
-	u, err := url.Parse(p.pt.finalGetURL())
-	if err != nil {
-		return nil, err
-	}
-	req := azcore.NewRequest(http.MethodGet, *u)
-	if err != nil {
-		return nil, err
-	}
-	resp, err := p.pipeline.Do(ctx, req)
-	if err != nil {
-		return nil, err
-	}
-	return p.handleResponse(resp)
+	return p.handleResponse(&azcore.Response{resp})
 }
 
 // ResumeToken generates the string token that can be used with the ResumeProductResultPagerPoller method
 // on the client to create a new poller from the data held in the current poller type
 func (p *productResultPagerPoller) ResumeToken() (string, error) {
-	if p.pt.hasTerminated() {
-		return "", errors.New("cannot create a ResumeToken from a poller in a terminal state")
-	}
-	js, err := json.Marshal(p.pt)
-	if err != nil {
-		return "", err
-	}
-	return string(js), nil
+	return p.pt.ResumeToken()
 }
 
 func (p *productResultPagerPoller) pollUntilDone(ctx context.Context, frequency time.Duration) (ProductResultPager, error) {
-	// initial check for a retry-after header existing on the initial response
-	if retryAfter := azcore.RetryAfter(p.pt.latestResponse().Response); retryAfter > 0 {
-		err := delay(ctx, retryAfter)
-		if err != nil {
-			return nil, err
-		}
-	}
-	// begin polling the endpoint until a terminal state is reached
-	for {
-		resp, err := p.Poll(ctx)
-		if err != nil {
-			return nil, err
-		}
-		if p.Done() {
-			break
-		}
-		d := frequency
-		if retryAfter := azcore.RetryAfter(resp); retryAfter > 0 {
-			d = retryAfter
-		}
-		err = delay(ctx, d)
-		if err != nil {
-			return nil, err
-		}
+	s := &productResultPager{}
+	resp, err := p.pt.PollUntilDone(ctx, frequency, p.pipeline, s)
+	if err != nil {
+		return nil, err
 	}
-	return p.FinalResponse(ctx)
+	return p.handleResponse(&azcore.Response{resp})
```

Comment:
Why did this change?

Reply:
Everything here is handled in PollUntilDone (including the call to FinalResponse on the armcore side), the only thing left to do is parse the final response we receive into the pager instance we want to return.

Reply:
I get it now, and we have to do this here because it's slightly different for pager-pollers.